tests/test_selective_validation.py

```diff
@@ -16,45 +16,100 @@ def __call__(self, *args, **kwargs):
 
 
 @pytest.mark.parametrize(
-    "shape, origin, domain",
+    "shape1, origin1, domain1, shape2, origin2, domain2",
     [
-        pytest.param((3, 3, 3), (0, 0, 0), (3, 3, 3), id="whole_array_valid"),
-        pytest.param((3, 3, 3), (0, 0, 0), (0, 0, 0), id="whole_array_invalid"),
-        pytest.param((4, 4, 4), (1, 2, 0), (2, 1, 3), id="some_valid"),
+        pytest.param(
+            (3, 3, 3),
+            (0, 0, 0),
+            (3, 3, 3),
+            (3, 3),
+            (0, 0),
+            (3, 3),
+            id="whole_array_valid",
+        ),
+        pytest.param(
+            (3, 3, 3),
+            (0, 0, 0),
+            (0, 0, 0),
+            (3, 3),
+            (0, 0),
+            (0, 0),
+            id="whole_array_invalid",
+        ),
+        pytest.param(
+            (4, 4, 4), (1, 2, 0), (2, 1, 3), (4, 4), (1, 0), (2, 3), id="some_valid"
+        ),
     ],
 )
 @pytest.mark.parametrize(
-    "all_arg_names, selective_arg_names",
+    "all_arg_names, selective_arg_names_shape1, selective_arg_names_shape2",
     [
-        pytest.param(["a"], [], id="no_selective_validation"),
-        pytest.param(["a"], ["a"], id="one_out_of_one_arg"),
-        pytest.param(["a", "b", "c"], ["b"], id="one_out_of_three_args"),
-        pytest.param(["a", "b", "c"], ["a", "b"], id="two_out_of_three_args"),
+        pytest.param(["a"], [], [], id="no_selective_validation"),
+        pytest.param(["a"], ["a"], [], id="one_out_of_one_arg"),
+        pytest.param(["a", "b", "c"], ["a"], [], id="one_shape1_arg"),
+        pytest.param(["a", "b", "c"], [], ["a"], id="one_shape2_arg"),
```

Comment:
```suggestion
        pytest.param(["a", "b", "c"], ["a"], [], id="one_shape1_arg"),
```
These two tests are identical from the standpoint of the tested code, shape1 and shape2 are implementation details of the test and not features of the tested API. you should be able to delete one case.

Reply:
Please merge this suggestion.